src/stability_sdk/client.py

```diff
@@ -234,6 +117,197 @@ def open_channel(host: str, api_key: str = None) -> generation_grpc.GenerationSe
     return generation_grpc.GenerationServiceStub(channel)
 
 
+class ApiEndpoint:
+    def __init__(self, stub, engine_id):
+        self.stub = stub
+        self.engine_id = engine_id
+
+class Api:
```

Comment:
I like what you've got going here. It feels like this class serves an extremely similar purpose to the StabilityInference class: we should probably role them together into a single thing or figure out what the separation of responsibilities/use-cases is so we can make their distinction clearer.

I'm personally inclined to just have one class since I think they're both basically serving the same purpose. Maybe the path forward is to stick a pin in the StabilityInference class, maybe add a deprecation warning alerting users to migrate in a forthcoming sdk release. Only thing that stands out to me rn that I'd want to change about the Api class is to separate request construction from request processing.

Reply:
yes ultimately i would like it returning placeholders so you can string together calls with the placeholder / named-buffers and then flush at some point before needing to do something with an actual result. this way can string the RPC requests into ChainGenerate calls.